pkg/agent/route/route_linux.go

```diff
@@ -1088,6 +1111,31 @@ func (c *Client) restoreIptablesData(podCIDR *net.IPNet,
 		c.writeEKSMangleRules(iptablesData)
 	}
 
+	if c.networkConfig.TrafficEncapMode == config.TrafficEncapModeHybrid {
+		writeLine(iptablesData, []string{
+			"-A", antreaPreRoutingChain,
+			"-m", "comment", "--comment", `"Antrea: restore fwmark from connmark for reply Egress packets from remote Pods"`,
+			"!", "-i", c.nodeConfig.GatewayConfig.Name, // Don't match packets from the local gateway to exclude request Egress packets.
+			"-m", "conntrack", "--ctstate", "ESTABLISHED", // Match packets from established connections.
+			"-m", "conntrack", "--ctdir", "REPLY", // Match reply packets.
+			// Match packets with conntrack mark.
+			"-m", "connmark", "--mark", fmt.Sprintf("%#08x/%#08x", types.EgressReturnViaTunMark, types.EgressReturnViaTunMark),
+			// Restore fwmark from the conntrack mark.
+			"-j", "CONNMARK", "--restore-mark", "--nfmask", fmt.Sprintf("%#08x", types.EgressReturnViaTunMark), "--ctmask", fmt.Sprintf("%#08x", types.EgressReturnViaTunMark),
+		}...)
+		writeLine(iptablesData, []string{
+			"-A", antreaPreRoutingChain,
+			"-m", "comment", "--comment", `"Antrea: persist connmark for the first Egress request packet from remote Pods"`,
+			"-i", c.nodeConfig.GatewayConfig.Name,
+			"!", "-s", podCIDR.String(), // Don't match packets from local Pods.
+			"-m", "conntrack", "--ctstate", "NEW", // Match the first packet.
+			// Match packets whose fwmark value (0–7) is non-zero. In the OVS pipeline, this mark is set for Egress connections.
+			"-m", "mark", "!", "--mark", fmt.Sprintf("%#08x/%#08x", 0, types.SNATIPMarkMask),
+			// Load EgressReturnViaTunMark into the conntrack mark so it persists for the lifetime of the Egress connection.
+			"-j", "CONNMARK", "--set-mark", fmt.Sprintf("%#08x/%#08x", types.EgressReturnViaTunMark, types.EgressReturnViaTunMark),
```

Comment:
I prefer a name like `EgressReturnToRemote`, as `ViaTun` is by OVS flows and not known to iptables rules.

Reply:
Updated to EgressReturnToRemote.